### ISSUE TITLE
KeyError: 'danger_level_tomorrow'

### ISSUE THREAD

**Issue by jonwaland** [exxamalte/home-assistant-custom-components-nsw-rural-fire-service-fire-danger#63]
```
2023-05-04 00:10:15.046 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 193, in _handle_refresh_interval
    await self._async_refresh(log_failures=True, scheduled=True)
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 341, in _async_refresh
    self.async_update_listeners()
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 142, in async_update_listeners
    update_callback()
  File "/home/homeassistant/.homeassistant/custom_components/nsw_rural_fire_service_fire_danger/entity.py", line 66, in _handle_coordinator_update
    self._update_state(self.coordinator.data[self._sensor_type])
KeyError: 'danger_level_tomorrow'
```

```
component version 0.16
Version	core-2023.4.6
Installation Type	Home Assistant Core
Development	false
Supervisor	false
Docker	false
User	homeassistant
Virtual Environment	true
Python Version	3.10.11
Operating System Family	Linux
Operating System Version	5.4.0-144-generic
CPU Architecture	x86_64
Timezone	Australia/Sydney
Configuration Directory	/home/homeassistant/.homeassistant
```

Forcing a reload of the component:
```
2023-05-04 10:37:44.143 ERROR (MainThread) [homeassistant.components.binary_sensor] Error while setting up nsw_rural_fire_service_fire_danger platform for binary_sensor
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/entity_platform.py", line 312, in _async_setup_platform
    await asyncio.gather(*pending)
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/entity_platform.py", line 455, in async_add_entities
    await asyncio.gather(*tasks)
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/entity_platform.py", line 731, in _async_add_entity
    await entity.add_to_platform_finish()
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/entity.py", line 822, in add_to_platform_finish
    await self.async_added_to_hass()
  File "/home/homeassistant/.homeassistant/custom_components/nsw_rural_fire_service_fire_danger/entity.py", line 59, in async_added_to_hass
    await self.async_update()
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 439, in async_update
    await self.coordinator.async_request_refresh()
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 200, in async_request_refresh
    await self._debounced_refresh.async_call()
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/debounce.py", line 89, in async_call
    await task
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 226, in async_refresh
    await self._async_refresh(log_failures=True)
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 341, in _async_refresh
    self.async_update_listeners()
  File "/srv/homeassistant/lib/python3.10/site-packages/homeassistant/helpers/update_coordinator.py", line 142, in async_update_listeners
    update_callback()
  File "/home/homeassistant/.homeassistant/custom_components/nsw_rural_fire_service_fire_danger/entity.py", line 66, in _handle_coordinator_update
    self._update_state(self.coordinator.data[self._sensor_type])
KeyError: 'danger_level_tomorrow'
```

**Comment by exxamalte** [comment]
Thanks for reporting this.
Looks like the XML feed isn't reporting the danger level for tomorrow. Never seen that before.
```
<District>
<Name>Far North Coast</Name>
<RegionNumber>1</RegionNumber>
<Councils>Ballina; Byron; Clarence Valley; Kyogle; Lismore; Richmond Valley; Tweed</Councils>
<DangerLevelToday>NONE</DangerLevelToday>
<FireBanToday>No</FireBanToday>
<FireBanTomorrow>No</FireBanTomorrow>
</District>
```

The JSON feed is reporting an empty string for tomorrow and two days after. I should catch all these cases, but probably need to somehow distinguish between "None"/"No Rating" vs. unpublished for that day.

**Comment by exxamalte** [comment]
This issue is now fixed in v0.17.
